core/eolearn/tests/test_utils/test_parsing.py

```diff
@@ -0,0 +1,263 @@
+    parser = FeatureParser(test_case.input)
+    assert parser.get_features() == test_case.features
+    assert parser.get_renamed_features() == test_case.renaming
+    assert parser.get_feature_specifications() == test_case.specifications
+
+
+@pytest.mark.parametrize("test_case", TEST_CASES_ELLIPSIS)
+def test_FeatureParser_Error(test_case: TestClass):
+    """Test failing when test_case.input contains ... because it has no meaning without EOPatch."""
+    parser = FeatureParser(test_case.input)
+    with pytest.raises(ValueError):
+        parser.get_features()
+    with pytest.raises(ValueError):
+        parser.get_renamed_features()
+
+
+@pytest.fixture(name="eopatch")
+def eopatch_fixture():
+    eopatch = EOPatch()
+    eopatch.data["bands"] = np.arange(2 * 3 * 3 * 2).reshape(2, 3, 3, 2)
+    eopatch.data["IS_VALID"] = np.arange(2 * 3 * 3 * 2).reshape(2, 3, 3, 2)
+    eopatch.data["CLP"] = np.arange(2 * 3 * 3 * 2).reshape(2, 3, 3, 2)
+    eopatch.data["CLM"] = np.arange(2 * 3 * 3 * 2).reshape(2, 3, 3, 2)
+    eopatch.bbox = BBox((1, 2, 3, 4), CRS.WGS84)
+    eopatch.mask["bands"] = np.arange(2 * 3 * 3 * 2).reshape(2, 3, 3, 2)
+    eopatch.mask["IS_VALID"] = np.arange(2 * 3 * 3 * 2).reshape(2, 3, 3, 2)
+    eopatch.mask["CLP"] = np.arange(2 * 3 * 3 * 2).reshape(2, 3, 3, 2)
+    eopatch.mask["CLM"] = np.arange(2 * 3 * 3 * 2).reshape(2, 3, 3, 2)
+    return eopatch
+
+
+@pytest.mark.parametrize("test_case", TEST_CASES_EMPTY + TEST_CASES + SPECIAL_CASES + TEST_CASES_ELLIPSIS)
+def test_FeatureParser_EOPatch(test_case: TestClass, eopatch: EOPatch):
+    parser = FeatureParser(test_case.input)
+    assert parser.get_features(eopatch) == test_case.features
+    assert parser.get_renamed_features(eopatch) == test_case.renaming
+
+
+@pytest.fixture(name="empty_subset_eopatch")
+def empty_eopatch_fixture():
+    eopatch = EOPatch()
+    eopatch.data["CLP"] = np.arange(2 * 3 * 3 * 2).reshape(2, 3, 3, 2)
+    eopatch.mask["bands"] = np.arange(2 * 3 * 3 * 2).reshape(2, 3, 3, 2)
+    return eopatch
+
+
+@pytest.mark.parametrize("test_case", TEST_CASES + TEST_CASES_ELLIPSIS)
+def test_FeatureParser_EOPatch_Error(test_case: TestClass, empty_subset_eopatch: EOPatch):
+    """Test failing when the test_case.input is not subset of EOPatch attributes."""
+    parser = FeatureParser(test_case.input)
+    with pytest.raises(ValueError):
+        parser.get_features(empty_subset_eopatch)
+    with pytest.raises(ValueError):
+        parser.get_renamed_features(empty_subset_eopatch)
+
+
+def test_FeatureParser_allowed_Error():
+    """Test failing when some features are not allowed."""
+    with pytest.raises(ValueError):
+        FeatureParser(
+            (
+                (
+                    (FeatureType.DATA, "bands", "new_bands"),
+                    (FeatureType.MASK, "bands", "new_bands"),
+                    (FeatureType.MASK, "CLP", "new_CLP"),
+                ),
+                (FeatureType.MASK),
```

Comment:
perhaps pass this as a keyword argument `allowed_feature_types=(FeatureType.MASK,)` to make tests more readable
currently i think the test passes but catches the wrong error

Reply:
done